src/kernels/MIOpenBatchNormBwdSpatial.cl

```diff
@@ -229,9 +229,19 @@ MIOpenBatchNormBwdSpatial(const __global _FLOAT* __restrict x_in,
 
 #elif(MIO_BN_VARIANT == 1)
 
+#ifndef MIO_LAYOUT_NHWC
+#define MIO_LAYOUT_NHWC 0
+#endif
```

Comment:
[Recommendation] Right now we define the default (0) but do not check if macro is correctly defined by MIOpen, so:
```suggestion
#ifndef MIO_LAYOUT_NHWC
#define MIO_LAYOUT_NHWC 0
#endif

#if (MIO_LAYOUT_NHWC != 0) || (MIO_LAYOUT_NHWC != 1)
#error MIO_LAYOUT_NHWC must be 0 or 1
#endif
```
or simply this (to enforce MIOpen to always correctly define this macro):
```suggestion
#if (MIO_LAYOUT_NHWC != 0) || (MIO_LAYOUT_NHWC != 1)
#error MIO_LAYOUT_NHWC must be 0 or 1
#endif
```
But this may be overkill of course :smile: 
If you like this, please make it everywhere, thanks.